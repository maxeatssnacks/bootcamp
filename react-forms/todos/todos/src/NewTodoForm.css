.NewTodoForm {
    margin-bottom: 20px;
}

.NewTodoForm form {
    display: flex;
    gap: 10px;
}

.NewTodoForm input {
    flex-grow: 1;
    padding: 10px;
    border: 1px solid #ddd;
    border-radius: 4px;
}

.NewTodoForm button {
    background-color: #4CAF50;
    color: white;
    border: none;
    padding: 10px 15px;
    border-radius: 4px;
    cursor: pointer;
    transition: background-color 0.3s;
}

.NewTodoForm button:hover {
    background-color: #45a049;
}