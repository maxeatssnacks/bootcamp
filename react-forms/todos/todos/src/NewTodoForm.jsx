import React, { useState } from "react";
import "./NewTodoForm.css";

const NewTodoForm = ({ addTodo }) => {
    const INITIAL_STATE = { task: '' };
    const [formData, setFormData] = useState(INITIAL_STATE);

    const handleChange = evt => {
        const { name, value } = evt.target;
        setFormData(formData => ({
            ...formData,
            [name]: value
        }));
    };

    const handleSubmit = evt => {
        evt.preventDefault();
        if (formData.task.trim()) {
            addTodo({ ...formData });
            setFormData(INITIAL_STATE);
        }
    };

    return (
        <div className="NewTodoForm">
            <form onSubmit={handleSubmit}>
                <input
                    type="text"
                    id="task"
                    name="task"
                    value={formData.task}
                    onChange={handleChange}
                    placeholder="Enter a task here"
                />
                <button type="submit">Add Task</button>
            </form>
        </div>
    );
};

export default NewTodoForm;