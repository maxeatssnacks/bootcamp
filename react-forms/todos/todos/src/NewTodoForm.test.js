import React from 'react';
import '@testing-library/jest-dom';
import { render, screen, fireEvent } from '@testing-library/react';
import NewTodoForm from './NewTodoForm';

// Smoke test
test('NewTodoForm renders without crashing', () => {
    render(<NewTodoForm addTodo={() => { }} />);
});

// Snapshot test
test('NewTodoForm matches snapshot', () => {
    const { asFragment } = render(<NewTodoForm addTodo={() => { }} />);
    expect(asFragment()).toMatchSnapshot();
});

// Critical business logic test
test('form submits correctly', () => {
    const mockAddTodo = jest.fn();
    render(<NewTodoForm addTodo={mockAddTodo} />);

    // Find the input and add button
    const input = screen.getByPlaceholderText('Enter a task here');
    const addButton = screen.getByText('Add Task');

    // Fill out the form and submit
    fireEvent.change(input, { target: { value: 'New Todo' } });
    fireEvent.click(addButton);

    // Check if the addTodo function was called with the correct argument
    expect(mockAddTodo).toHaveBeenCalledWith({ task: 'New Todo' });

    // Check if the input is cleared after submission
    expect(input.value).toBe('');
});