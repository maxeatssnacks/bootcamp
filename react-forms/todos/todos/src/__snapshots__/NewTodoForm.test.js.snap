// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewTodoForm matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="NewTodoForm"
  >
    <form>
      <input
        id="task"
        name="task"
        placeholder="Enter a task here"
        type="text"
        value=""
      />
      <button
        type="submit"
      >
        Add Task
      </button>
    </form>
  </div>
</DocumentFragment>
`;
